### PR TITLE
Webpack

### LINKED ISSUE #79: Add webpack for JS

### PR DESCRIPTION
**Pull request by lejoe**
Fixes #79 

Finally went for declarative dependencies.
In short you need to require your component scripts in the main files.

Also for vendor scripts you need to add them in the /gulp/tasks/scripts so that they are added to the vendor file.

Coming in separate PRs (TODO):
- production build optimisation

### CODE REVIEW

**Comment by lejoe** [comment]
Ready to review @lucalanca @peterbraden

**Comment by meodai** [comment]
So if I get you, we need to copy files from NPM to /gulp/tasks/scripts ? If we want them to be in the vendor file?

**Comment by lejoe** [comment]
@meodai nope, you need to add them here: https://github.com/ginetta/skeleton/pull/101/files#diff-9d68bb3ecce3a165761fdcb2b7966095R24

**Comment by meodai** [comment]
aha so you ment: "/gulp/tasks/scripts.js"

File: src/elements/header/script.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+'use strict';
+var document, window;
+(function(){
```
Comment (peterbraden) [inline src/elements/header/script.js]: Why are we using this syntax? Webpack will wrap the code for us. This seems backward.

File: src/scripts/main.js [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
+
+// This is how you require a component from node_modules
+// Make sure to add it to the vendors in /gulp/tasks/scripts
+var $ = require('jquery');
+
 var document, window;
 (function(){
   // TODO: add your JS here
-}(document, window));
+}(document, window, $));
```
Comment (peterbraden) [inline src/scripts/main.js]: Let's definitely use the CommonJS style

File: gulp/tasks/scripts.js [hunk 3]
```diff
@@ -1,14 +1,45 @@
...
+  var skeletonRoot = config.basePaths.root;
+  var srcRoot = config.basePaths.src;
 
   var task = function () {
-    return gulp.src(srcFiles)
+    return gulp.src(scriptsFiles)
       .pipe($.eslint({envs: ['browser']}))
       .pipe($.eslint.format())
+      .pipe(gulpWebpack({
+        debug: true, //TODO improve this one we have env depending builds
```
Comment (peterbraden) [inline gulp/tasks/scripts.js]: Why not just use `process.env.NODE_ENV != 'production' ` for now?

File: src/elements/header/script.js (hunk 1 above)
Comment (lejoe) [inline src/elements/header/script.js]: valid

File: src/scripts/main.js (hunk 2 above)
Comment (lejoe) [inline src/scripts/main.js]: jup

File: gulp/tasks/scripts.js (hunk 3 above)
Comment (lejoe) [inline gulp/tasks/scripts.js]: Cause I don't want half temp solutions for the skeleton.
Will address this in #84

Comment (peterbraden) [inline gulp/tasks/scripts.js]: Ok.

File: gulp/tasks/scripts.js [hunk 4]
```diff
@@ -1,14 +1,45 @@
...
+        entry: {
+          main: srcFiles,
+          // Add modules you want to load from vendors here so they are put in a seperate file
+          vendor: ['jquery']
```
Comment (lucalanca) [inline gulp/tasks/scripts.js]: this doesn't feel right. basically we're saying that we will always use jquery.

I would prefer to have a vendor.js file instead and don't leak the vendor dependencies into the gulp configuration.

**Comment by lejoe** [comment]
Ok @ginetta/frontend I think this is ready

**Comment by lucalanca** [comment]
LGTM.

**Comment by peterbraden** [comment]
LGTM

**Pull request merged**